system/jlib/jmetrics.hpp

```diff
@@ -341,4 +395,25 @@ class jlib_decl ScopedGaugeDecrementer
     int64_t amount;
 };
 
+
+class jlib_decl ScopedCountIncrement
+{
+public:
+    explicit ScopedCountIncrement(CounterMetric &_pCount, int64_t _amount=1)
+    : count{_pCount}, amount{_amount}
+    { }
+    ScopedCountIncrement(const ScopedGaugeDecrementer&) = delete;
+    ScopedCountIncrement(ScopedGaugeDecrementer&) = delete;
```

Comment:
is this useful?  I think it is implied by copy constructor on the  line above.  Same for the second assignment.  Did you mean to use &&?  I'm not sure it is likely to catch real world problems.

Reply:
After some further research, the non const copy constructor and assignment operator should be sufficient. Move copy constructors and assignment are not automatically generated, so they were left out. However, since we are deleting the non move copy constructor, best to ensure they are deleted as well.